### DISCUSSION TITLE
Not seeing floating window with g:lsp_diagnostics_float_cursor=1

### DISCUSSION THREAD

**Discussion by asagent7** [prabirshrestha/vim-lsp#1040] [Q&A] [answered]
I am setting up vim-lsp as well as asyncomplete based off default configurations suggested in the README. I can see the diagnostic messages via virtual text if I enable it but not in a floating window if I use lsp_diagnostics_float_cursor.
Using neovim 0.4.2, and I can see floating windows for :LspHover, so floating windows are supported.

```
" Register servers for vim-lsp
au User lsp_setup call lsp#register_server({
    \ 'name': 'pyls',
    \ 'cmd': {server_info->['pyls']},
    \ 'allowlist': ['python'],
    \ 'workspace_config': {'pyls': {'plugins': {'jedi_completion': {'fuzzy': v:true}}}},
    \})

" Keymap for vim-lsp
function! s:on_lsp_buffer_enabled() abort
    setlocal omnifunc=lsp#complete
    setlocal signcolumn=yes
    if exists('+tagfunc') | setlocal tagfunc=lsp#tagfunc | endif

    let g:lsp_format_sync_timeout = 1000
    let g:lsp_diagnostics_signs_delay = 2000
    let g:lsp_diagnostics_virtual_text_delay = 2000
    let g:lsp_diagnostics_virtual_text_enabled = 0
    let g:lsp_diagnostics_float_cursor = 1
    "let g:lsp_documentation_float = 0
    "let g:lsp_log_verbose = 1
    "let g:lsp_log_file = expand('$XDG_DATA_HOME/nvim/vim-lsp.log')
    
endfunction

augroup lsp_install
    au!
    " call s:on_lsp_buffer_enabled only for languages that has the server registered.
    autocmd User lsp_buffer_enabled call s:on_lsp_buffer_enabled()
augroup EN

"Other settings for asyncomplete
let g:asyncomplete_min_chars=3
let g:asyncomplete_auto_completeopt = 0
set completeopt=menuone,noinsert,noselect,preview
autocmd! CompleteDone * if pumvisible() == 0 | pclose | endif
```

**Comment by dhduvall** [comment]
Hah. I didn't see this before I filed #1081, but I've been running into the same thing.

**Comment by prabirshrestha** [comment, marked as answer]
` let g:lsp_diagnostics_float_cursor = 1` flag is checked on very early so make sure to move it outside of `on_lsp_buffer_enabled` function.

These are global files and I would put it right before `register_server` calls.

**Reply by asagent7** [reply]
This worked. Thanks.

**Comment by rvyhvn** [comment]
@prabirshrestha 
I use Neovim with vim-lsp and Vimscript config.

When setting let g:lsp_diagnostics_float_cursor = 1, diagnostics appear in a floating popup — but it’s not focusable or scrollable.

In coc.nvim, the diagnostic popup can be clicked/focused and even scrolled.

Is it possible to replicate this with vim-lsp on Neovim?

I’m looking for a way to make the popup focusable, or display diagnostics in a floating window that I can interact with using normal keys.
